### PR TITLE
Make imshow return a viewer

### PR DESCRIPTION
**Pull request by sofroniewn**
# Description

This very short PR reverts our main imshow api so that it returns a viewer, similar to the api before the refactor and of our last version of PyPi as discussed with @jni in #108 . In order to achieve this the window now gets appended as a property on the viewer. This is not necessarily a permanent solution but will allow us to push a new release without breaking the old api. Further discussion of the topic can happen on the issue.

## Type of change

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# References

# How has this been tested?

- [x] ran `examples/layers.ipynb`
- [x] ran `examples/demo.py`

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: gui/elements/_window.py [hunk 1]
```diff
@@ -33,6 +33,7 @@ def __init__(self, viewer, show=True):
 
         self.viewer.events.status.connect(self._status_changed)
         self.viewer.events.help.connect(self._help_changed)
+        self.viewer.window = self
```
Comment (kevinyamauchi) [inline gui/elements/_window.py]: self.viewer.window.viewer.window... :P

Comment (sofroniewn) [reply gui/elements/_window.py]: :-( indeed, less than ideal, but the only way i can see how to do it. Also this is possible in matplotlib, but doesn't seem to cause them problems (as far as i know)

**Review by kevinyamauchi — APPROVED** [review]
Thanks, @sofroniewn ! This looks good to me. I agree that it's better to return `viewer`.

It looks like the linux and windows tests are failing, but it looks like an issue with the cirrus setup (exit code 134 - memory issue?).

**Review by kne42 — CHANGES_REQUESTED** [review]
I would just add the created windows to a list to keep reference to them as done in [`v0.0.1`](https://github.com/Napari/napari-gui/blob/v0.0.1/napari_gui/util/misc.py#L106).

Also, to keep this PR slightly less trivial I would also unify `viewer.imshow` and `viewer.add_image` by moving the code from `imshow` into `ImageLayer`'s constructor and adding a new parameter to `add_to_viewer` that allows you to specify aliases for the function.

File: gui/elements/_window.py (hunk 1 above)
Comment (sofroniewn) [reply gui/elements/_window.py]: this is now fixed, thanks to @kne42!

**Comment by sofroniewn** [comment]
Thanks @kne42 I have now switched to using a windows list to prevent garbage collection, but still allow for imshow to return a viewer. Maybe this addresses some concerns of @AhmetCanSolak?

**Review by kne42 — APPROVED** [review]
In the interest of keeping things moving quickly I’ll address my picky comments in a follow-up PR. This looks acceptable to me as-is.

**Pull request merged**